CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.10.2)
+
+# where to find our CMake modules
+set(CMAKE_MODULE_PATH "$ENV{CARMA_OPT_DIR}/cmake")
+# Required for std::lock and syntax
+set(CMAKE_CXX_STANDARD 17)
+
+# Needed so CMake can find the vendored PROJ4 module file. Th FindPROJ4.cmake
+# module file can be removed if we upgrade to a more recent PROJ version. See
+# https://github.com/usdot-fhwa-stol/carma-platform/issues/2139 for updates.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+project(carma-geodetic
+    DESCRIPTION "CARMA geodetic library"
+    HOMEPAGE_URL https://github.com/usdot-fhwa-stol/carma-geodetic-lib
+	VERSION 0.0.1
+    LANGUAGES CXX
+)
+
+find_package(Threads)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
I don't see anywhere in the code that requires us to link specifically with `pthread`. This should be removed.

Reply:
Its required by linking to the test executable created by the https://github.com/usdot-fhwa-stol/carma-builds/blob/cd6e4ce3d443539ec4fd598d1c41225d79ecdcc4/cmake/Testing.cmake . To me it seems if it is really required there we should include it there. Not sure I understand this dependencies but have seen it before for creating test executable. Reading up on it to try and educate myself. Without this I get the following error though:
```
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libgtest.a(gtest-all.cc.o): undefined reference to symbol 'pthread_getspecific@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

Reply:
I think this is a bug in `Testing.cmake`. It should be linking against `GTest::gtest_main` instead of `gtest`. Can you make that change locally and see if that resolves it?

Reply:
Works and added the following PR to fix it (https://github.com/usdot-fhwa-stol/carma-builds/pull/27/)